packages/next/src/server/app-render/app-render-prerender-utils.ts

```diff
@@ -0,0 +1,226 @@
+/**
+ * This utility function is extracted to make it easier to find places where we are doing
+ * specific timing tricks to try to schedule work after React has rendered. This is especially
+ * import at the moment because Next.js uses the edge builds of React which use setTimeout to
+ * schedule work when you might expect that something like setImmediate would do the trick.
+ *
+ * Long term we should switch to the node versions of React rendering when possible and then
+ * update this to use setImmediate rather than setTimeout
+ *
+ * A shorter term work around would be to patch React to use setImmediate instead of setTimeout
+ * in the edge builds since this might also avoid setTimeout throttling.
```

Comment:
we're already doing this as of https://github.com/vercel/next.js/pull/68952, so maybe it makes sense to make this an x-ref instead of a theoretical statement?